### PR TITLE
Merge the js-parser and parser files

### PR DESCRIPTION
**Pull request by robertsipka**

### CODE REVIEW

File: jerry-core/parser/js/js-parser.cpp [hunk 1]
```diff
@@ -1817,8 +1817,8 @@ parser_free_literals (parser_list_t *literal_pool_p) /**< literals */
  *
  * @return compiled code
  */
-ecma_compiled_code_t *
-parser_parse_script (const uint8_t *source_p, /**< valid UTF-8 source code */
+static ecma_compiled_code_t *
+parse_script (const uint8_t *source_p, /**< valid UTF-8 source code */
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser.cpp]: Parameters are not aligned.
And I would prefer parser_parse_source

File: jerry-core/parser/js/js-parser.cpp [hunk 2]
```diff
@@ -1817,8 +1817,8 @@ parser_free_literals (parser_list_t *literal_pool_p) /**< literals */
...
                      size_t size, /**< size of the source code */
```
Comment (LaszloLango) [inline jerry-core/parser/js/js-parser.cpp]: wrong indentation

File: jerry-core/parser/js/js-parser.h [hunk 3]
```diff
@@ -128,8 +128,23 @@ typedef struct
   parser_line_counter_t column;                       /**< column where the error occured */
 } parser_error_location;
 
-/* Note: source must be a valid UTF-8 string. */
-ecma_compiled_code_t * parser_parse_script (const uint8_t *, size_t, int, parser_error_location *);
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser.h]: The comment should be kept.

**Comment by robertsipka** [comment]
Thanks the comments. Fixed.

File: jerry-core/parser/js/js-parser.h (hunk 3 above)
Comment (zherczeg) [inline jerry-core/parser/js/js-parser.h]: This one is not fixed. After that, LGTM

**Comment by robertsipka** [comment]
Updated.

**Comment by zherczeg** [comment]
LGTM

**Comment by LaszloLango** [comment]
Landed (b0bdf0ebf4ee89181a55e1c8dbfb49d5b314a633)

**Pull request closed without merging**